src/int/specialized_div_rem/asymmetric.rs

```diff
@@ -0,0 +1,179 @@
+/// Creates unsigned and signed division functions optimized for dividing integers with the same
+/// bitwidth as the largest operand in an asymmetrically sized division. For example, x86-64 has an
+/// assembly instruction that can divide a 128 bit integer by a 64 bit integer if the quotient fits
+/// in 64 bits. The 128 bit version of this algorithm would use that fast hardware division to
+/// construct a full 128 bit by 128 bit division.
+#[macro_export]
+macro_rules! impl_asymmetric {
+    (
+        $unsigned_name:ident, // name of the unsigned division function
+        $signed_name:ident, // name of the signed division function
+        $zero_div_fn:ident, // function called when division by zero is attempted
```

Comment:
As a general rule we don't need to check for zero here. The API contract for the division functions is that a zero check will have already been done beforehand. It is not necessary to check for zero again here and it's fine if we just call `unreachable_unchecked` or trigger a CPU trap in that case.

Reply:
I think we should keep `$zero_div_fn` so that we can define it to be `::abort` or `unreachable_unchecked` in one place